brayns/io/loaders/volume/VolumeLoader.cpp

```diff
@@ -207,50 +247,40 @@ std::vector<std::string> RawVolumeLoader::getSupportedExtensions() const
-
-bool MHDVolumeLoader::isSupported(const std::string &, const std::string &extension) const
-{
-    return extension == "mhd";
-}
-
-std::vector<ModelDescriptorPtr> MHDVolumeLoader::importFromBlob(const Blob &, const LoaderProgress &, Scene &) const
+std::vector<std::shared_ptr<Model>> MHDVolumeLoader::importFromBlob(const Blob &blob, const LoaderProgress &callback)
+    const
 {
-    throw std::runtime_error("Volume loading from blob is not supported");
+    (void)blob;
+    (void)callback;
+    throw std::runtime_error("MHD Volume loading from blob is not supported");
 }
 
-std::vector<ModelDescriptorPtr>
-    MHDVolumeLoader::importFromFile(const std::string &filename, const LoaderProgress &callback, Scene &scene) const
+std::vector<std::shared_ptr<Model>> MHDVolumeLoader::importFromFile(
+    const std::string &filename,
+    const LoaderProgress &callback) const
 {
-    std::string volumeFile = filename;
-    const auto mhd = parseMHD(filename);
-
-    // Check all keys present
-    for (const auto key : {"ObjectType", "DimSize", "ElementSpacing", "ElementType", "ElementDataFile"})
-        if (mhd.find(key) == mhd.end())
-            throw std::runtime_error("Missing key " + std::string(key));
-
-    if (mhd.at("ObjectType") != "Image")
-        throw std::runtime_error("Wrong object type for mhd file");
+    auto mhd = MhdParser::parse(filename);
+    MhdRequiredKeys::check(mhd);
 
-    const auto dimensions = parseVec3<int32_t>(mhd.at("DimSize"), [](const auto &s) { return stoi(s); });
-    const auto spacing = parseVec3<double>(mhd.at("ElementSpacing"), [](const auto &s) { return stod(s); });
-    const auto type = dataTypeFromMET(mhd.at("ElementType"));
-
-    std::filesystem::path path = mhd.at("ElementDataFile");
-    if (!path.is_absolute())
+    auto &objectType = mhd["ObjectType"];
+    if (objectType != "Image")
     {
-        auto basePath = std::filesystem::path(filename).parent_path();
-        path = std::filesystem::canonical(basePath / path);
+        throw std::runtime_error("Wrong object type for mhd file");
     }
-    volumeFile = path.string();
 
-    RawVolumeLoaderParameters params;
+    auto rawDimensions = std::string_view(mhd["DimSize"]);
+    auto rawSpacing = std::string_view(mhd["ElementSpacing"]);
+    auto dimensions = Parser::extractToken<Vector3ui>(rawDimensions);
+    auto spacing = Parser::extractToken<Vector3f>(rawSpacing);
+    auto dataType = MhdDataTypeConverter::convert(mhd["ElementType"]);
+    auto volumeFilePath = MhdRawFilePathResolver::resolve(mhd["ElementDataFile"], filename);
+
+    auto params = RawVolumeLoaderParameters();
     params.dimensions = dimensions;
     params.spacing = spacing;
-    params.type = type;
+    params.data_type = dataType;
 
-    return RawVolumeLoader().importFromFile(volumeFile, callback, params, scene);
+    return RawVolumeLoader().importFromFile(volumeFilePath, callback, params);
 }
 
 std::string MHDVolumeLoader::getName() const
```

Comment:
I know it is the old name but we have different case for almost all loader names ^^.
Like SONATA loader, mesh, mhd-volume,...

Reply:
I dont understand this one